src/boot/efi/util.h

```diff
@@ -52,6 +52,10 @@ static inline void *xrealloc(void *p, size_t old_size, size_t new_size) {
         return t;
 }
 
+#define xrealloc_extra_k_items(type, p, current_size, k) xrealloc(p, \
+                current_size == 0 ? 0 : (current_size + 1) * sizeof((type *)), \
+                (current_size + k + 1) * sizeof((type *)))
```

Comment:
i don't like the amount of double-eval/side-effects. Can't we do better, maybe with a static inline helper?

also misses a couple of () to avoid unexpected operator resolution order

Reply:
Here's a suggestion with static inline helper, we lose the `type*` on the extra_k_items, but seems fine.